### DISCUSSION TITLE
Kafka POD nodeName as container env

### DISCUSSION THREAD

**Discussion by vijayrajah** [strimzi/strimzi-kafka-operator#4868] [General]
Hello,

We are using Datadog to monitor our Kafka cluster. I want to tag all the JMX metrics with appropriate tags to ensure we can query them. One of the current issues is, we want to tag the metrics with pod name. This will enable us to track kafka metrics over time (as container ip can change & Pod name will stay consistent)

I see a enhancement was requested in #4023 , which was closed without any action.

Can we please re-open it & hopefully add the support for downward API?

-Vijay

**Comment by scholzj** [comment]
I'm not sure I follow the requirement. How will the environment variable be consumed? To collect the metrics from JMX, your monitoring system needs to know where to connect and so it already knows the name of the pod, or? If it reads the variable from the Kubernetes API and Pod YAML, it again already knows the pod name. That is at least how it works with Prometheus - it knows what it is doing and just adds the label if needed.
